### ISSUE TITLE
Title for the bug

### ISSUE THREAD

**Issue by darko-vidakovic-ch** [firebase/firebase-js-sdk#7807]
### Operating System

macOS 13.5.2 

### Browser Version

Google Chrome 119.0.6045.159

### Firebase SDK Version

10.5.2

### Firebase SDK Product:

Auth

### Describe your project's tooling

A React frontend, using typescript, swc and webpack, served through a NodeJS server.

### Describe the problem

Our project allows, besides Google OAuth and email/password authentication, custom SAML2 integrations.

Our sign in process is triggered with the `signInWithPopup` function. Some customers have SAML flows with a couple of redirects, which seem to trigger the `auth/popup-closed-by-user` error.

After investigating I've found that it happens in the following function:

```ts
  private pollUserCancellation(): void {
    const poll = (): void => {
      if (this.authWindow?.window?.closed) {
        // Make sure that there is sufficient time for whatever action to
        // complete. The window could have closed but the sign in network
        // call could still be in flight. This is specifically true for
        // Firefox or if the opener is in an iframe, in which case the oauth
        // helper closes the popup.
        this.pollId = window.setTimeout(() => {
          this.pollId = null;
          this.reject(
            _createError(this.auth, AuthErrorCode.POPUP_CLOSED_BY_USER)
          );
        }, _Timeout.AUTH_EVENT);
        return;
      }

      this.pollId = window.setTimeout(poll, _POLL_WINDOW_CLOSE_TIMEOUT.get());
    };

    poll();
  }
```
where `this.authWindow?.window?.closed` is `true`, probably due to the redirects.

Unfortunately, switching to the `signInWithRedirect` flow is not an option at this moment.

### Steps and code to reproduce issue

- Setup a simple app that uses the `signInWithPopup` flow
- Setup a custom SAML2 provider which has multiple redirects in it's authentication flow
- Try and sign in using that provider
- See that the `auth/popup-closed-by-user` error is shown even though the popup is still open.

**Comment by zmitry** [comment]
Omg +100, we hit the same issue at our company as well.

**Comment by renkelvin** [comment]
Thanks @darko-vidakovic-ch , we'd like to know more information to better understand this issue.

Is it a single popup window that redirects multiple times, or multiple popup window? (I assume a former but to confirm)
How long does it take before the the sign in flow error out? like how many seconds?

It would be great if you can provide a link that can reproduce this issue, or a video recording. Thanks!

**Comment by renkelvin** [comment]
Just discussed with team that we have a 8-second timeout limit and I think that's why you receive the error here. It seems it stuck at the meta sign-in page for a while, I wonder if anything can be improved there.

**Comment by darko-vidakovic-ch** [comment]
Yes, it's that exact timeout that kicks in here. It's rather a poll interval to check the `window.closed` property of the auth window.

The problem here is that in the first 8 seconds a couple of redirects happen within the popup prior to actually starting the sign in process from the user's perspective. Because of these redirects, the SDK concludes that the auth popup is closed even though it's not, and in 8 seconds we get this error. Since the actual customer's SAML auth flows are not managed by us, we cannot do anything to prevent such redirects, and expect more cases like this.

Basically, what I was implying was that the logic behind deducing whether the popover is closed, which is invoked every 8 seconds, probably does not work as expected with redirects.

To mitigate this issue in production we've deployed a patch that overrides the 8 second timeout for that particular customer and sets it to something much longer, but it's far from a fix.

**Comment by darko-vidakovic-ch** [comment]
> Hey @darko-vidakovic-ch! We faced the same issue in our project. Could you, please, share what exactly did you do to make it work?

Hey, unfortunately, we're still living with the issue 😞 To mitigate the problem to some extent, since only a subset of our customers has such redirects, we did a post-install patch of the Firebase auth SDK that allows us to inject a timeout that is more practical for our use case, but the root issue persists. At this point it's either that or switching to a redirect-based auth flow, but that one comes with its own set of caveats (3rd party storage access).

@renkelvin do you know by any chance if a fix for this is somewhere in the pipeline? Maintaining these patches is cumbersome since we'll need to regenerate them every time we want to upgrade.

Thanks!

**Comment by Timi-Leyin** [comment]
still no update?😫
